Calibration/HcalCalibAlgos/test/HcalLaserTest.cc

```diff
@@ -0,0 +1,217 @@
+  TH1F                                    *h_hb1_, *h_hb2_, *h_hb3_, *h_hb4_;
+  TH1F                                    *h_hb5_, *h_hf1_, *h_hf2_;
+};
+
+
+HcalLaserTest::HcalLaserTest(const edm::ParameterSet& config) :
+  inputTokenHBHE_( consumes<HBHEDigiCollection>( config.getParameter<edm::InputTag>("InputHBHE") ) ),
+  inputTokenHF_( consumes<QIE10DigiCollection>( config.getParameter<edm::InputTag>("InputHF") ) ),
+  ioTokenUMN_( consumes<HcalUMNioDigi>( config.getParameter<edm::InputTag>("UMNioDigis") )),
+  minFracDiffHBHELaser_(config.getParameter<double>("minFracDiffHBHELaser")),
+  minFracHFLaser_(config.getParameter<double>("minFracHFLaser")),
+  minADCHBHE_(config.getParameter<int>("minADCHBHE")),
+  minADCHF_(config.getParameter<int>("minADCHF")),
+  testMode_(config.getUntrackedParameter<bool>("testMode", false)),
+  eventsByType_(),
+  passedEventsByType_() { 
+
+  usesResource(TFileService::kSharedResource);
+}
+
+HcalLaserTest::~HcalLaserTest() { }
+ 
+void HcalLaserTest::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("InputHBHE",edm::InputTag("source"));
+  desc.add<edm::InputTag>("InputHF",edm::InputTag("source"));
+  desc.add<edm::InputTag>("UMNioDigis",edm::InputTag("UMNioDigis"));
+  desc.add<int>("minADCHBHE",10);
+  desc.add<int>("minADCHF",10);
+  desc.add<double>("minFracDiffHBHELaser",0.3);
+  desc.add<double>("minFracHFLaser",0.3);
+  desc.addUntracked<bool>("testMode",false);
+  descriptions.add("hcalLaserTest",desc);
+}
+
+void HcalLaserTest::analyze(edm::Event const& iEvent, 
+			    edm::EventSetup const& iSetup) {
+  
+  edm::Handle<HBHEDigiCollection>     hbhe_digi; 
+  iEvent.getByToken(inputTokenHBHE_,  hbhe_digi);
+
+  edm::Handle<QIE10DigiCollection>    hf_digi; 
+  iEvent.getByToken(inputTokenHF_,    hf_digi);
+
+  // Count digis in good, bad RBXes.  ('bad' RBXes see no laser signal)
+  double badrbxfracHBHE(0), goodrbxfracHBHE(0), rbxfracHF(0);
+  int NbadHBHE  = HcalBadLaserChannels::badChannelsHBHE();
+  int NgoodHBHE = 2592*2-NbadHBHE;  // remaining HBHE channels are 'good'
+  int NallHF    = 864*4;
+  int eventType = 0;
+  int laserType = 0;
+
+  // Verify proper functioning on calibration stream events, 
+  // where the laser firing location "laserType"
+  // is tagged by the UMNio board for eventType==14
+
+  edm::Handle<HcalUMNioDigi> cumn;
+  iEvent.getByToken(ioTokenUMN_, cumn);
+
+  eventType = static_cast<int>(cumn->eventType());
+  laserType = static_cast<int>(cumn->valueUserWord(0));
+
+  if (eventType == 14) eventsByType_.at(laserType)++;
+
+  if (testMode_)
+    edm::LogVerbatim("HcalLaserTest") 
+      << "hbhe digi collection size: " << hbhe_digi->size() << "\n"
+      << "hf digi collection size: " << hf_digi->size() << "\n"
+      << "Event type: " << eventType 
+      << " Laser type: " << laserType << std::endl;
+
+  for (auto hbhe : *hbhe_digi) {
```

Comment:
Please merge this now - I shall see the change in a later PR. This PR is pending for a long time and I like to use some of the code used here for some other application